vpr/src/place/analytic_placer.h

```diff
@@ -0,0 +1,249 @@
+ * solution is not legal. It generally has many blocks overlapping with one another, and the blocks may be on
+ * incompatible physical tiles. To legalize this solution, a geometric partitioning and spreading technique, introduced
+ * in SimPL, is used (@see cut_sreader.h). This completes the upper-bound placement.
+ *
+ * After the completion of 1 iteration of lower-bound & upper-bound placement, artificial pseudo connections are created
+ * between each block and its target location in the legalized overlap-free placement. When the mathematical system is
+ * again formulated and solved, the pseudo connections pull blocks towards their target locations, which tends to reduce
+ * overlaps in the placement.
+ *
+ * This process of formulating the system, solving, and legalizing is repeated until sufficiently good placement is
+ * acquired. Currently the stopping criterion is 15 iterations without improvement in total_hpwl.
+ *
+ *
+ * Parameters to tweak & things to try out
+ * =======================================
+ * Currently the QoR and runtime of AP+quench combination is slightly worse than SA. See PR #1504 for comparison.
+ * The following parameters/things can be tweaked to find the best configuration:
+ *
+ * * Stopping criteria			when to stop AP iterations, see (AnalyticPlacer::ap_place())
+ * * PlacerHeapCfg.alpha		anchoring strength of pseudo-connection
+ * * PlacerHeapCfg.beta			overutilization factor (@see CutSpreader::SpreaderRegion.overused())
+ * * PlacerHeapCfg.timingWeight	implement timing in AP (@see AnalyticPlacer::build_equations())
+ * * PlacerHeapCfg.criticality	same as above
+ * * Interaction with SA:
+ * 	 * init_t					Initial temperature of annealer after AP (currently init_t = 0)
+ * 	 * quench inner_num			how much swapping in quenching to attemp
+ * 	 * quench_recompute_limit	frequency of criticality update in quenching to improve quench results
+ *
+ * @cite SimPL
+ * Original analytic placer with cut-spreading legalizing was intended for ASIC design, proposed in SimPL.
+ * SimPL: An Effective Placement Algorithm, Myung-Chul Kim, Dong-Jin Lee and Igor L. Markov
+ * http://www.ece.umich.edu/cse/awards/pdfs/iccad10-simpl.pdf
+ *
+ * @cite HeAP
+ * FPGA adaptation of SimPL, targeting FPGAs with heterogeneous blocks located at discrete locations.
+ * Analytical Placement for Heterogeneous FPGAs, Marcel Gort and Jason H. Anderson
+ * https://janders.eecg.utoronto.ca/pdfs/marcelfpl12.pdf
+ *
+ * @cite nextpnr
+ * An implementation of HeAP, which the cut-spreader and legalizer here is based off of. Implementation details
+ * have been modified for the architecture and netlist specification of VTR, and better performance.
+ * nextpnr -- Next Generation Place and Route,  placer_heap, David Shah <david@symbioticeda.com>
+ * https://github.com/YosysHQ/nextpnr
+ */
+
+#    include "vpr_context.h"
+#    include "timing_place.h"
+#    include "PlacementDelayCalculator.h"
+
+typedef signed long int wirelen_t;
```

Comment:
Comment what this is for (wirelength) and why you use it.  This is still only a 32 bit integer but that should be enough for integral wirelength of an FPGA.